### ISSUE TITLE
fetch does not work with local urls on Now

### ISSUE THREAD

**Issue by uunomee** [thgh/now-sapper#8]
The following is the code...

In **.src/routes/index.svelte**
```
<script context="module"> 
    export async function preload({ params, query }) { 
        const res = await this.fetch(`/data`);
        const data = await res.json();
        return data;//{"title":data.title};
    }
</script>   
<script>  
export let title;
</script>  
<h1>Great success! - {title}</h1>  
```
created **.src/routes/data.js** file and have the following
```
const superagent = require('superagent'); 
// GET /data
export function get(req, res, next) { 
  superagent.get("https://external/api/json/get/path/xyz")
    .type("json")
    .accept("json")
    .query(req.query)
    .pipe(res);
}
```

1.  **{title}** in the above code only works when we click on other tabs and back but not when we refresh the index.svelte page
2.  The same code works fine locally ( localhost ) but not when we deploy to now ( production )

**Comment by uunomee** [comment]
Thanks! to @thgh for all the support and help. His initial suggestion was to fetch external url directly
In **.src/routes/index.svelte**   
`const res = await this.fetch("https://external/api/json/get/path/xyz"); `
instead of using local
`const res = await this.fetch("/data");`

**Comment by thgh** [comment]
I thought of another workaround 😛 

```html
<script context="module">
  export async function preload() {
    const baseURL =
      !process.browser && process.env.NOW_REGION
        ? 'https://your-project.now.sh'
        : ''
    const res = await this.fetch(baseURL + `/data`)
    const data = await res.json()
    return data //{"title":data.title};
  }
</script>
```

This will run a function in a function, not really efficient but it might work.

**Comment by uunomee** [comment]
Thanks! the direct fetch( external URL ) worked, will try this too :)

**Comment by jnv** [comment]
Thanks for the tips! You can also use `host` property in the `preload` function so you don't need to hardcode domain URL. This is the workaround I have arrived to:

```js
  export function preload({host, params, query}) {
    let baseUrl = ''
    if (!process.browser && process.env.NOW_REGION) {
      baseUrl = `https://${host}`
    }
    
    return this.fetch(`${baseUrl}/data`)
    // ...
}
```

**Comment by thgh** [comment]
Fix has been released in v0.39.0! Thanks @fpaboim
